src/transformers/generation/utils.py

```diff
@@ -1917,8 +1928,8 @@ def generate(
                 raise ValueError("assisted generate is only supported for batch_size = 1")
             if not model_kwargs["use_cache"]:
                 raise ValueError("assisted generate requires `use_cache=True`")
-            if generation_config.cache_implementation == "static":
-                raise ValueError("assisted generate is not supported with `static_cache`")
+            if generation_config.cache_implementation in ["static", "hybrid", "sliding_window"]:
+                raise ValueError("assisted generate is not supported with Static cachc classes`")
```

Comment:
```suggestion
                raise ValueError("assisted generate is not supported with Static cache classes`")
```

Reply:
But I think it is supported for `hybrid`, as local gemma and gemma2 release was about that

Reply:
It was failing with size mismatch and Hybrid is basically as Static, I'll see if that was a bugu in gemma/tests or it is indeed not supported

Reply:
@ArthurZucker it is not supported with hybrid cache, for the same reasons as static cache isn't [it needs functions to roll back the cache, which we haven't implemented there].

Gemma 2 + assisted generation works because we force gemma 2 to use dynamic cache there :)

Reply:
Ahh okay got it thanks